lib/traject/sul/normalized_id.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'arclight/exceptions'
+
+module Sul
+  ##
+  # A utility class to normalize identifiers
+  class NormalizedId
+    # The Arclight to_field directive sends kwargs (:title and :repository)
+    # to this class and we use the id and title to form the collection id
+    def initialize(id, **kwargs)
+      @id = id
+      @title = kwargs[:title]
+    end
+
+    def to_s
+      normalize
+    end
+
+    private
+
+    attr_reader :id, :title
+
+    def normalize
+      parameterized_id = parameterize(id)
+      parameterized_title = parameterize(title)
+
+      # De-duplicate edge cases where id has been generated from title due to missing EAD ID
+      # E.g. don't return Université de Toulouse_universite-de-toulouse
+      # Instead, return only universite-de-toulouse
+      return parameterized_id if parameterized_id == parameterized_title
+
+      normalized_id = [parameterized_id, parameterized_title].compact.join('_')
```

Comment:
Maybe? (and then remove the parameterize method?)
```suggestion
      # De-duplicate edge cases where id has been generated from title due to missing EAD ID
      # E.g. don't return Université de Toulouse_universite-de-toulouse
      # Instead, return only universite-de-toulouse
      normalized_id = [id, title].map do |string|
        string&.truncate_words(5, omission: '')&.parameterize
      end.compact.uniq.join('_')
```